python_scripts/ensemble_gradient_boosting.py

```diff
@@ -126,29 +144,22 @@ def generate_data(n_samples=50):
 target_test_predicted_residuals = tree_residuals.predict(data_test)
 
 # %%
-sns.scatterplot(x=data_train["Feature"], y=residuals, color="black", alpha=0.5)
-line_predictions = plt.plot(
-    data_test["Feature"], target_test_predicted_residuals, "--"
+handles, ax = plot_decision_tree_with_residuals(
+    residuals,
+    target_train_predicted_residuals,
+    target_test_predicted_residuals,
 )
-
-# plot the residuals of the predicted residuals
-for value, true, predicted in zip(
-    data_train["Feature"], residuals, target_train_predicted_residuals
-):
-    lines_residuals = plt.plot([value, value], [true, predicted], color="red")
-
-plt.legend(
-    [line_predictions[0], lines_residuals[0]],
-    ["Fitted tree", "Residuals"],
-    bbox_to_anchor=(1.05, 0.8),
-    loc="upper left",
-)
-_ = plt.title("Prediction of the previous residuals")
+legend_labels = [
```

Comment:
can we pass the legend labels and title as parameter as well like this the `plot_` function takes really care about all plotting.

Reply:
And we don't need to define the story about the `handles`

Reply:
I thought that making the helper accept legend labels and titles may be a bit too much of customization.

Reply:
Fair enough.